gobblin-utility/src/main/java/gobblin/util/AvroUtils.java

```diff
@@ -136,7 +142,7 @@
    */
   public static GenericRecord convertRecordSchema(GenericRecord record, Schema newSchema) throws IOException {
     if (checkReaderWriterCompatibility(newSchema, record.getSchema()).getType() != COMPATIBLE) {
-      LOG.warn("Record schema not compatible with writer schema. Converting record schema to writer schema may fail.");
+      LOG.debug("Record schema not compatible with writer schema. Converting record schema to writer schema may fail.");
```

Comment:
Because if schemas are not compatible, the conversion may fail, but may also succeed. So if a job has a large number of calls of this method, that warn message may flood the log.